### ISSUE TITLE
axios 1.1.3 can't use in a library

### ISSUE THREAD

**Issue by c4tc4tc4t** [axios/axios#5154]
### Describe the issue

Hi, i've created a library and it was working fine, but after the update to 1.1.3 it started to give me a bunch of the following error :
"BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
        - add a fallback 'resolve.fallback: { "stream": require.resolve("stream-browserify") }'
        - install 'stream-browserify'
If you don't want to include a polyfill, you can use an empty module like this:
        resolve.fallback: { "stream": false }","

I'm currently using 1.1.2 and it's working fine, but since i'm not the only one that uses the library, i'm afraid of this issue, how can i fix this? i already tried to upgraded the version of both(library and project) to 1.1.3 and doesn't work, the only thing that worked for me until now was setting everything back to 1.1.2. Help!
 

### Axios Version

1.1.3

### Additional Library Versions

```bash
angular 14.0.0
```

**Comment by ajmas** [comment]
You may want to look at you NodeJS version and what ES version you are targeting. It sounds like something is outdated in your tool chain?

**Comment by AlbertAZ1992** [comment]
It seems like the change of browser enrty with built cjs file affect more than expected?
When axios is used to create a library as above, the dist file can not be process with tool chain, while nodejs core modules (stream, https, etc) are not be polyfilled in dist cjs file... 

https://github.com/axios/axios/compare/v1.1.2...v1.1.3#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R9 
![image](https://user-images.githubusercontent.com/8591672/196590661-7bdb1ecb-74a3-4730-bd1e-76f84d86b680.png)

**Comment by jasonsaayman** [comment]
Ok, will have to have a look can i be given the environment details NPM/Node versions etc

**Comment by c4tc4tc4t** [comment]
node v16.17.1 , it's a typescript library that i use in my angular projects, the version of Angular of my projects are all 14.0.0

**Comment by povilass** [comment]
Browsers use modules/or will transpile module maybe just add it "import": "./dist/esm/axios.js",?

**Comment by ajmas** [comment]
I'd be concerned about needing to do a deep import of anything in a package.

@c4tc4tc4t Are you able to create a minimal Angular project that demonstrates the issue? It may be easier for other contributors to investigate based on this?

**Comment by c4tc4tc4t** [comment]
https://github.com/c4tc4tc4t/axioslibtest

there u go, make sure to install on a angular project to test, you can use the function checkImportStatus to test it, the error will occur when you run ng serve

**Comment by c4tc4tc4t** [comment]
You can also clone my lib and change the version of axios to "1.1.2" and you will see that everything works fine.

**Comment by ajmas** [comment]
> https://github.com/c4tc4tc4t/axioslibtest
> 
> there u go @ajmas , make sure to install on a angular project to test, you can use the function checkImportStatus to test it, the error will occur when you run ng serve

I have never really worked with Angular, so I am not sure about the remaining recipe to be able to run this.

**Comment by c4tc4tc4t** [comment]
ok, running some tests i found out that the problem is in the type AxiosResponseHeaders that comes from axios, did u guys change something with this type in this update?

**Comment by c4tc4tc4t** [comment]
still not figured out, acctually the lib that i created does not contain the error, my bad sry, but the actual lib i cant post it

**Comment by c4tc4tc4t** [comment]
Ok, so what you have to do is create a Angular 14.0.0 project, then run "npm install https://github.com/c4tc4tc4t/poketeste" , i will install my typescript library that contains only one function, install axios in the Angular project and import the lib "import * as poke from 'axiosteste'", axiosteste is the name of my lib, try a console.log(poke.pokemon()); , if u run ng serve, you will get my error, waiting for some help on this, and of course if u change the version of axios on my lib and the project back to 1.1.2 everything works fine!

**Comment by povilass** [comment]
Well this changed
![image](https://user-images.githubusercontent.com/8037810/197388718-0c26b55f-02d4-45d5-942d-8477fae638c6.png)

Kinda destroyed what's use for browser and node platform because  

axios\dist\node\axios.cjs

Got this 
![image](https://user-images.githubusercontent.com/8037810/197388798-1ee599f7-7a33-4da3-ba34-9cc5b737f203.png)

And axios\dist\axios.js got this 
![image](https://user-images.githubusercontent.com/8037810/197388872-feeaf549-befd-4ead-8773-ab7720dae5c9.png)

So this change must be reverted....

**Comment by krzaczek** [comment]
We have the same problem with our React app. We have been using axios since 0.19 and 1.1.3 breaks our webpack build. Reverting to 1.1.2 fixes the problem for
now but this needs to be fixed.

**Comment by c4tc4tc4t** [comment]
ok, thank you!

**Comment by krutoo** [comment]
i have same issue, can someone tell how to fix that wit 1.1.3 (without downgrade)?

**Comment by jlombardo-tableau** [comment]
yes, we need an actual fix for this, not to just revert to 1.1.2 in our code.

**Comment by ajmas** [comment]
Given this issue isn't really fixed and instead only has a downgrade as workaround, we should probably keep this ticket open?

**Comment by KubaLorenc** [comment]
It works well with 1.2.0 version, but is broken again in 1.2.2

**Comment by ajmas** [comment]
@KubaLorenc does it break in the same way as before the fix? If not, can you describe the failure mode?

**Comment by mevannir** [comment]
I have the same problem when used in a library I am developing and after importing and using it in a CRA app. Its displays pollyfill errors, I am using 1.3.5 v of axios. Is there any solution for this

**Comment by ajmas** [comment]
Can you share the errors? Did you try adding the polyfill it is complaining about, as a package dependency?

**Comment by mevannir** [comment]
@ajmas Yes I added all the complaining polyfills, some errors were resolved but still there are more errors displaying as below. When I remove any use of axios in my library and imported the updated version to the sample app then the pollyfill errors are gone. Therefore I identified these polyfill errors come due to use of Axios (1.3.5)

```
If you want to include a polyfill, you need to:
	- add a fallback 'resolve.fallback: { "stream": require.resolve("stream-browserify") }'
	- install 'stream-browserify'
```
	
The above same error is displaying for stream-http, https-browserify, fs, tty-browserify, os-browserify, browserify-zlib.
